src/net.cpp

```diff
@@ -1375,6 +1376,47 @@ void CConnman::SocketEvents(std::set<SOCKET> &recv_set, std::set<SOCKET> &send_s
         if(FD_ISSET(hSocket, &fdsetError))
             error_set.insert(hSocket);
 }
+#else
+void CConnman::SocketEvents(std::set<SOCKET> &recv_set, std::set<SOCKET> &send_set, std::set<SOCKET> &error_set)
+{
+    std::set<SOCKET> recv_select_set, send_select_set, error_select_set;
+    GenerateSelectSet(recv_select_set, send_select_set, error_select_set);
+
+    std::vector<struct pollfd> pollfds;
+    struct pollfd pollfd;
+    memset(&pollfd, 0, sizeof(struct pollfd));
+
+    for (SOCKET hSocket : recv_select_set) {
+        pollfd.fd = hSocket;
+        pollfd.events = POLLIN;
+        pollfds.push_back(pollfd);
+    }
+
+    for (SOCKET hSocket : send_select_set) {
+        pollfd.fd = hSocket;
+        pollfd.events = POLLOUT;
+        pollfds.push_back(pollfd);
+    }
+
+    for (SOCKET hSocket : error_select_set) {
+        pollfd.fd = hSocket;
+        pollfd.events = POLLERR|POLLHUP;
+        pollfds.push_back(pollfd);
+    }
+
+    if(poll(&pollfds[0], pollfds.size(), 50) < 0)
```

Comment:
nit: Make the 50 a named constant along with `timeout.tv_usec` in the `select` impl.